### PR TITLE
Add properties

### PR DESCRIPTION
**Pull request by KrechAA**
properties is added

### CODE REVIEW

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    public WordController(WordService wordService, WordRepository wordRepository, IndexRepository indexRepository) {
+        this.wordService = wordService;
+        this.wordRepository = wordRepository;
+        this.indexRepository = indexRepository;
+    }
+
+
+    @PostMapping(value = "/word/create")
+    public ResponseEntity<?> saveOneWord(@RequestBody RestWordObject restWordObject) {
+        WordObject wordObject = new WordObject(restWordObject.getWord(), restWordObject.getFirstLetter());
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Extract lines 33-35 to the Service method marked with @Transactional annotation. Controller->@Transactional Service->Repositoriy

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    @Autowired
+    public WordController(WordService wordService, WordRepository wordRepository, IndexRepository indexRepository) {
+        this.wordService = wordService;
+        this.wordRepository = wordRepository;
+        this.indexRepository = indexRepository;
+    }
+
+
+    @PostMapping(value = "/word/create")
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Remove /create

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    public ResponseEntity<?> saveOneWord(@RequestBody RestWordObject restWordObject) {
+        WordObject wordObject = new WordObject(restWordObject.getWord(), restWordObject.getFirstLetter());
+        indexRepository.deleteAll();
+        wordRepository.save(wordObject);
+        return new ResponseEntity<>(HttpStatus.CREATED);
+    }
+
+
+    @DeleteMapping(value = {"/word/delete"})
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Remove /delete

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            if (wordRepository.findByName(name) == null) {
+                return new ResponseEntity<>(HttpStatus.OK);
+            } else {
+                return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+            }
+        }
+    }
+
+
+    @PutMapping(value = "/word/update")
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Remove /update

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    @DeleteMapping(value = {"/word/delete"})
+    public ResponseEntity<?> delete(@RequestParam String name) {
+
+        if (wordRepository.findByName(name) == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        } else {
+            indexRepository.deleteAll();
+            wordRepository.deleteByName(name);
+            if (wordRepository.findByName(name) == null) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: There is no need to check word deletion.

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        }
+    }
+
+
+    @PutMapping(value = "/word/update")
+    public ResponseEntity<?> update(@RequestParam String oldword, @RequestBody RestWordObject restNewWordObject) {
+        WordObject newWordObject = new WordObject(restNewWordObject.getWord(), restNewWordObject.getFirstLetter());
+        WordObject oldWordObject = wordRepository.findByName(oldword);
+        if (oldWordObject == null) {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Not_found status is better here

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        } else {
+            indexRepository.deleteAll();
+            newWordObject.setId(oldWordObject.getId());
+            wordRepository.update(newWordObject.getName(), newWordObject.getFirstLetter(), newWordObject.getId());
+            if (wordRepository.findByName(newWordObject.getName()) == null) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: There is no need to check word update.

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    @PutMapping(value = "/word/update")
+    public ResponseEntity<?> update(@RequestParam String oldword, @RequestBody RestWordObject restNewWordObject) {
+        WordObject newWordObject = new WordObject(restNewWordObject.getWord(), restNewWordObject.getFirstLetter());
+        WordObject oldWordObject = wordRepository.findByName(oldword);
+        if (oldWordObject == null) {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        } else {
+            indexRepository.deleteAll();
+            newWordObject.setId(oldWordObject.getId());
+            wordRepository.update(newWordObject.getName(), newWordObject.getFirstLetter(), newWordObject.getId());
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: use save(newWordObject) instead

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    @Autowired
+    public WordController(WordService wordService, WordRepository wordRepository, IndexRepository indexRepository) {
+        this.wordService = wordService;
+        this.wordRepository = wordRepository;
+        this.indexRepository = indexRepository;
+    }
+
+
+    @PostMapping(value = "/word/create")
+    public ResponseEntity<?> saveOneWord(@RequestBody RestWordObject restWordObject) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: First check, if the word already exists. If it is throw illegalargex with description of error.

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+                return new ResponseEntity<>(HttpStatus.OK);
+            } else {
+                return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+            }
+        }
+    }
+
+
+    @PutMapping(value = "/word/update")
+    public ResponseEntity<?> update(@RequestParam String oldword, @RequestBody RestWordObject restNewWordObject) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: First check, if the word already exists. If it is throw illegalargex with description of error.

File: src/main/java/com/krech/botv3/domain/Indexkey.java [hunk 11]
```diff
@@ -30,13 +30,14 @@ public int hashCode() {
         return Objects.hash(firstChar, otherChars);
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/domain/Indexkey.java]: Looks as invalid equals implementation

File: src/main/java/com/krech/botv3/domain/rest/request/RestWordObject.java [hunk 12]
```diff
@@ -0,0 +1,28 @@
+package com.krech.botv3.domain.rest.request;
+
+public class RestWordObject {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/domain/rest/request/RestWordObject.java]: rename to WordResponse

File: src/main/java/com/krech/botv3/repository/IndexRepository.java [hunk 13]
```diff
@@ -1,18 +1,18 @@
...
+@Repository
+public interface IndexRepository extends CrudRepository<IndexObject, Integer> {
 
-   public void addNewIndexAndWords(Indexkey key, Set<String> newSetOfWords);
 
-    public Set <String> getWords(Indexkey key);
+    List<IndexObject> findByFirstLetter(String str);
 
-    public Map<Indexkey, Set<String>> getAll ();
+    IndexObject findByFirstLetterAndOtherLetter(String firstLetter, String otherLetters);
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/repository/IndexRepository.java]: OtherLetters in method name

File: src/main/java/com/krech/botv3/repository/WordRepository.java [hunk 14]
```diff
@@ -1,10 +1,35 @@
...
+@Repository
+public interface WordRepository extends CrudRepository<WordObject, Integer> {
+
+
+    List<WordObject> findByFirstLetter(String c);
+    WordObject findByName(String str);
+
+
+
+    @Transactional
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/repository/WordRepository.java]: @Transactional should be placed on the service level

File: src/main/java/com/krech/botv3/service/WordService.java [hunk 15]
```diff
@@ -1,24 +1,32 @@
...
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.*;
+import java.util.stream.Collectors;
 
 import static java.lang.Character.isUpperCase;
+
 @Service
+@Transactional
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/service/WordService.java]: Check if you really need all methods to be transactional. If it is, then it is correct, no problem

File: src/main/java/com/krech/botv3/service/WordService.java [hunk 16]
```diff
@@ -30,46 +38,40 @@ public WordService(IndexRepository indexRepository, WordRepository wordRepositor
      */
     public List<String> readWordsFromFile() throws IOException {
         List<String> list = new ArrayList<>();
-        Path path = Path.of("c:\\Users\\krech\\Downloads\\freqrnc2011.txt");
+        Path path = Path.of("c:\\Users\\krecha\\Downloads\\freqrnc2011.txt");
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/service/WordService.java:43]: It would be nice to create controller method to upload file from client and store data from it to DB.
See example https://stackoverflow.com/questions/39037049/how-to-upload-a-file-and-json-data-in-postman

Lets do this in separate Pull request

File: src/main/java/com/krech/botv3/service/WordService.java [hunk 17]
```diff
@@ -108,81 +104,106 @@ void loadWords() throws IOException {
...
+                        newSetOfWordObject.add(fromNewSet);
+                    }
+                }
+                indexObject.getSetOfWords().addAll(newSetOfWordObject);
+                indexRepository.save(indexObject);
             }
+
+
         } else {
             System.out.println("первая буква не большая");
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/service/WordService.java]: throw exeption

File: src/main/java/com/krech/botv3/service/WordService.java [hunk 18]
```diff
@@ -108,81 +104,106 @@ void loadWords() throws IOException {
...
     }
 
     /**
      * @param chars список запрошенных букв первая буква списка - первая буква слова
      * @param words список слов найденных по запрошенным буквам
      */
-    void saveIndex(char[] chars, Set<String> words) {
+    public void saveIndex(char[] chars, Set<WordObject> words) {
 
         if (isUpperCase(chars[0])) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/service/WordService.java]: invert if condition to make code more readable

File: src/main/java/com/krech/botv3/service/WordService.java [hunk 19]
```diff
@@ -108,81 +104,106 @@ void loadWords() throws IOException {
...
-                indexRepository.getWords(indexkey).addAll(words);//TODO заменить addAll на add в цикле.
+
+                Set<WordObject> existingWords = new HashSet<>(indexObject.getSetOfWords());
+
+                Set<WordObject> newSetOfWordObject = new HashSet<>();
+
+                for (WordObject fromNewSet : words) {
+                    if (!existingWords.contains(fromNewSet)) {
+
+                        newSetOfWordObject.add(fromNewSet);
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/service/WordService.java]: You can add words directly to indexObject.getSetOfWords() without newSetOfWordObject collection

File: src/test/java/com/krech/botv3/WordServiceTest.java [hunk 20]
```diff
@@ -16,56 +21,44 @@
...
+        WordObject fenixyatina = new WordObject("Фениксятина", "Ф");
 
-        setWithoutNumber.add("Феникс");
-        setWithoutNumber.add("Фениксятина");
-        set1.add("Феникс");
-        set1.add("Фениксятина");
-        set1.add("Фениксятиноподобный");
-        set1.add("Фениксблаблабла");
-        set1.add("Фениксовость");
+        NoSuchElementException thrown = Assertions.assertThrows(NoSuchElementException.class, () -> when(wordService.searchIndex(str.toCharArray())).thenReturn(null));
```
Comment (amkurakin32) [inline src/test/java/com/krech/botv3/WordServiceTest.java]: You should mock repositories, not wordService methods. Please don't mix assertions and mocks

File: src/test/java/com/krech/botv3/WordServiceTest.java [hunk 21]
```diff
@@ -75,68 +68,64 @@ void searchWordsForClientTest(){
...
+        setOfWordObjects.add(fenix);
+        setOfWordObjects.add(fenixyatina);
+
+        IndexObject fenixIndex = new IndexObject();
+        fenixIndex.setOtherLetter("еник");
+        fenixIndex.setFirstLetter("Ф");
+        fenixIndex.setWords(setOfWordObjects);
+
+
+        when(indexRepository.findByFirstLetter(String.valueOf(chars[0]))).thenReturn(List.of(fenixIndex));
```
Comment (amkurakin32) [inline src/test/java/com/krech/botv3/WordServiceTest.java]: wrap String.valueOf(chars[0])) to eq(String.valueOf(chars[0])))

File: pom.xml [hunk 22]
```diff
@@ -70,6 +78,36 @@
...
+
+        <!-- https://mvnrepository.com/artifact/org.hibernate/hibernate-entitymanager -->
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-entitymanager</artifactId>
+
+        </dependency>
+
+        <!-- https://mvnrepository.com/artifact/org.json/json -->
+        <dependency>
```
Comment (amkurakin32) [inline pom.xml]: Check if we need these two dependencies

File: src/main/java/com/krech/botv3/controller/BotController.java [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+
+    @GetMapping(value = "/bot/search")
+    public ResponseEntity<List<String>> read(@RequestParam String letters) {
+        final List<String> words = wordService.searchWordsForClient(letters);
+
+        return words != null
+                ? new ResponseEntity<>(words, HttpStatus.OK)
+                : new ResponseEntity<>(HttpStatus.NOT_FOUND);
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/BotController.java]: Return NO_CONTENT

File: src/main/java/com/krech/botv3/controller/RestResponseEntityExceptionHandler.java [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.ServletWebRequest;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
+
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/RestResponseEntityExceptionHandler.java:15]: clean empty lines

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+    @PostMapping(value = "/words")
+    public ResponseEntity<?> saveOneWord(@RequestBody WordResponse wordResponse) {
+        if (wordRepository.findByName(wordResponse.getWord()) != null) {
+            throw new IllegalArgumentException("This word already exist");
+        }
+        return wordService.saveOneWord(wordResponse);
+    }
+
+
+    @DeleteMapping(value = {"/words"})
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Remove {} brackets

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+    @Autowired
+    public WordController(WordService wordService, WordRepository wordRepository, IndexRepository indexRepository) {
+        this.wordService = wordService;
+        this.wordRepository = wordRepository;
+        this.indexRepository = indexRepository;
+    }
+
+
+    @PostMapping(value = "/words")
+    public ResponseEntity<?> saveOneWord(@RequestBody WordResponse wordResponse) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Controller.saveOneWord should accept WordRequest without id and respond with WordResponse with id

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 27]
```diff
@@ -0,0 +1,69 @@
...
+        if (wordRepository.findByName(wordResponse.getWord()) != null) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Move this if statement inside wordService.saveOneWord

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 28]
```diff
@@ -0,0 +1,69 @@
...
+            throw new IllegalArgumentException("This word already exist");
+        }
+        return wordService.saveOneWord(wordResponse);
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: wordService.saveOneWord Should accept String word, String firstChar and return wordObject.

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 29]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+
+    @DeleteMapping(value = {"/words"})
+    public ResponseEntity<?> delete(@RequestParam String name) {
+
+        if (wordRepository.findByName(name) == null) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: move code inside new wordService method.

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 30]
```diff
@@ -0,0 +1,69 @@
...
+        } else {
+            indexRepository.deleteAll();
+            wordRepository.deleteByName(name);
+            return new ResponseEntity<>(HttpStatus.OK);
+        }
+    }
+
+
+    @PutMapping(value = "/words")
+    public ResponseEntity<?> update(@RequestParam String oldword, @RequestBody WordResponse restNewWordObject) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Controller.update should accept WordRequest without id and respond with ResponseEntity 'WordResponse' with id

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 31]
```diff
@@ -0,0 +1,69 @@
...
+        WordObject oldWordObject = wordRepository.findByName(oldword);
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Move all code inside wordService new method "public WordObject update(String oldWord, String newWord, String newWordFirstLetter)

File: src/main/java/com/krech/botv3/domain/rest/request/WordResponse.java [hunk 32]
```diff
@@ -0,0 +1,28 @@
+package com.krech.botv3.domain.rest.request;
+
+public class WordResponse {
+
+    private String word;
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/domain/rest/request/WordResponse.java]: Add id to response object. And create Request object without id

File: src/main/java/com/krech/botv3/service/WordService.java [hunk 33]
```diff
@@ -30,46 +40,39 @@ public WordService(IndexRepository indexRepository, WordRepository wordRepositor
...
     public List<String> readWordsFromFile() throws IOException {
         List<String> list = new ArrayList<>();
-        Path path = Path.of("c:\\Users\\krech\\Downloads\\freqrnc2011.txt");
+        Path path = Path.of("c:\\Users\\krecha\\Downloads\\freqrnc2011.txt");
         return list = Files.readAllLines(path);
 
     }
 
+
+    public ResponseEntity<?> saveOneWord(WordResponse wordResponse) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/service/WordService.java]: saveOneWord Should return wordObject from wordRepository.save(wordObject)

return wordRepository.save(wordObject);

File: src/test/java/com/krech/botv3/WordServiceTest.java [hunk 34]
```diff
@@ -16,56 +21,44 @@
...
-        set1.add("Фениксблаблабла");
-        set1.add("Фениксовость");
+        NoSuchElementException thrown = Assertions.assertThrows(NoSuchElementException.class, () -> when(wordService.searchIndex(str.toCharArray())).thenReturn(null));
 
-        indexStorage.put(indexkey,setWithoutNumber);
-        indexStorage.put(indexkey1,set1);
+        assertNull(thrown.getMessage());
 
-        when(inMemoryIndexRepositoryImpl.getAll()).thenReturn(indexStorage);
+        when(wordService.searchInWords(str.toCharArray())).thenReturn(List.of(fenix, fenixyatina));
```
Comment (amkurakin32) [inline src/test/java/com/krech/botv3/WordServiceTest.java]: You should mock repositories, not wordService methods.

File: src/main/java/com/krech/botv3/controller/WordController.java (hunk 28 above)
Comment (KrechAA) [reply src/main/java/com/krech/botv3/controller/WordController.java]: can we accept WordRequest?

File: src/main/java/com/krech/botv3/controller/WordController.java (hunk 31 above)
Comment (KrechAA) [reply src/main/java/com/krech/botv3/controller/WordController.java]: can we use WordRequest and WordResponse objects in this method?

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 35]
```diff
@@ -0,0 +1,53 @@
...
+
+
+    @Autowired
+    public WordController(WordService wordService) {
+        this.wordService = wordService;
+    }
+
+
+    @PostMapping(value = "/words")
+    public ResponseEntity<?> saveOneWord(@RequestBody WordRequest wordRequest) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Change returning type to ResponseEntity<WordResponse>

File: src/main/java/com/krech/botv3/controller/WordController.java [hunk 36]
```diff
@@ -0,0 +1,53 @@
...
+
+    @DeleteMapping(value = "/words")
+    public ResponseEntity<?> delete(@RequestParam String name) {
+        wordService.deleteOneWord(name);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+
+    @PutMapping(value = "/words")
+    public ResponseEntity<?> update(@RequestParam String oldword, @RequestBody WordRequest wordRequest) {
```
Comment (amkurakin32) [inline src/main/java/com/krech/botv3/controller/WordController.java]: Change returning type to ResponseEntity<WordResponse>

File: src/test/java/com/krech/botv3/WordServiceTest.java [hunk 37]
```diff
@@ -16,56 +21,45 @@
...
-        indexStorage.put(indexkey1,set1);
+        when(indexRepository.findByFirstLetter(eq(String.valueOf(str.charAt(0))))).thenReturn(listOfIndexObject);
+        when(wordRepository.findByFirstLetter(eq(String.valueOf(str.charAt(0))))).thenReturn(List.of(fenix, fenixyatina));
 
-        when(inMemoryIndexRepositoryImpl.getAll()).thenReturn(indexStorage);
 
-        List <String> words = wordService.searchWordsForClient(str);
+        List<String> words = wordService.searchWordsForClient(str);
 
         assertEquals(expected.size(), words.size());
```
Comment (amkurakin32) [inline src/test/java/com/krech/botv3/WordServiceTest.java:59]: It would be nice to initialize the "expected" array right before assertion. It will make code more readable. Also since java 11 you can do initialization in modern style: List<String> expected = List.of("Фениксятина", "Феникс");

File: src/test/java/com/krech/botv3/WordServiceTest.java [hunk 38]
```diff
@@ -75,68 +69,64 @@ void searchWordsForClientTest(){
...
 
-        HashMap<Indexkey, Set<String>> indexStorage = new HashMap<>();
+        when(indexRepository.findByFirstLetter(eq(String.valueOf(chars[0])))).thenReturn(List.of(fenixIndex));
 
-        when(inMemoryIndexRepositoryImpl.getAll()).thenReturn(indexStorage);
-//        when(indexkey.getFirstChar()).thenReturn('Ф');
-   //     when(indexkey.getOtherChars()).thenReturn("еник");
-    //    when(wordService.searchInWords(eq(chars))).thenReturn(List.of("Феникс", "Фениксятина"));
-    //    when(wordRepository.getWordsByChar(eq('Ф'))).thenReturn(List.of("Феникс", "Фуфайка", "Фениксятина"));
+        result = wordService.searchIndex(chars);
```
Comment (amkurakin32) [inline src/test/java/com/krech/botv3/WordServiceTest.java]: Move "result" variable declaration to initialization line.
